certificate-manager/byo-root.mdx

```diff
@@ -0,0 +1,168 @@
+
+```shell
+step certificate sign \
+   --not-after 87600h \
+   --profile intermediate-ca \
+   --path-len 0 \
+   csr-9e63a172-EXAMPLE-5b0c7.csr root_ca.crt root_ca_key
+```
+
+### Active Directory Certificate Services
+
+Use the `certreq` tool.
+
+```shell
+certreq -submit -attrib "CertificateTemplate:SubCA" csr-9e63a172-EXAMPLE-5b0c7.csr intermediate.crt
+```
+
+### AWS Certificate Manager Private CA
+
+You can now use the following python script that uses issue-certificate to process the CSR:
+
+```python nocopy
+import boto3
+import sys
+
+AWS_CA_ARN = '[YOUR_PRIVATE_CA_ARN]'
+
+csr = ''.join(sys.stdin.readlines())
+
+client = boto3.client('acm-pca')
+response = client.issue_certificate(
+    CertificateAuthorityArn=AWS_CA_ARN,
+    Csr=csr,
+    SigningAlgorithm='SHA256WITHRSA',
+    TemplateArn='arn:aws:acm-pca:::template/SubordinateCACertificate_PathLen1/V1',
+    Validity={
+        'Value': 5,
+        'Type': 'YEARS'
+    }
+)
+print(f"Creating certificate with ARN {response['CertificateArn']}...", file=sys.stderr, end='')
+waiter = client.get_waiter('certificate_issued')
+waiter.wait(
+    CertificateAuthorityArn=AWS_CA_ARN,
+    CertificateArn=response['CertificateArn']
+)
+print('done.', file=sys.stderr)
+response = client.get_certificate(
+   CertificateArn=response['CertificateArn'],
+   CertificateAuthorityArn=AWS_CA_ARN
+)
+print(response['Certificate'])
+```
+
+To run it, fill in the ARN of your CA and run:
+
+```shell-session nocopy
+$ python issue_certificate.py < csr-9e63a172-EXAMPLE-5b0c7.csr > intermediate.crt
+```
+
+### Hashicorp Vault
+
+You will need [`jq`](https://jqlang.github.io/jq/) installed, to extract the certificate PEM programmatically.
+
+Run:
+
+```
+INTERMEDIATE_CA_TTL="87600h"
+vault write -format=json pki/root/sign-intermediate \
+	csr=@csr-9a63a172-EXAMPLE-5b0c7.csr \
+	format=pem_bundle \
+	ttl=$INTERMEDIATE_CA_TTL | jq -r .data.certificate > intermediate.crt
+```
+
+### OpenSSL
+
+```shell-session nocopy
+openssl ca -config [ROOT_CA_CONFIG_FILE] \
+  -extensions v3_intermediate_ca \
+  -days 3650 -notext -md sha512 \
+  -in csr-9e63a172-EXAMPLE-5b0c7.csr \
+  -out intermediate.crt
+  ```
+
+### CFSSL
+
+For CFSSL you'll need a signing profile that specifies a 10-year expiry:
+``` shell-session nocopy
+$ cat > ca-smallstep-config.json <<EOF
+{
+  "signing": {
+    "profiles": {
+      "smallstep": {
+        "expiry": "87600h",
+        "usages": ["signing"]
+      }
+    }
+  }
+}
+EOF
+```
+
+Now use that config to sign the intermediate certificate:
+
+``` shell-session nocopy
+$ cfssl sign -ca ca.pem \
+    -ca-key ca-key.pem \
+    -config ca-smallstep-config.json \
+    -profile smallstep
+    -csr intermediate.csr | cfssljson -bare
+```
+
+This process will yield a signed intermediate.crt certificate (or cert.pem for CFSSL).
+
+## Conclusion
+
+You can now upload your signed certificate to Smallstep, and we'll create your new CA.
```

Comment:
I think it would be nice to expand the conclusion a little bit to touch on the different PKI's that we just went through, a note that this can be done with any PKI (not just these), and the reasoning/value-add for why people should do this.